components/script/dom/htmlinputelement.rs

```diff
@@ -81,7 +86,7 @@ pub struct HTMLInputElement {
     // https://html.spec.whatwg.org/multipage/#concept-input-value-dirty-flag
     value_dirty: Cell<bool>,
 
-    // TODO: selected files for file input
+    filelist: DOMRefCell<Option<JS<FileList>>>,
```

Comment:
Having read doc on `MutNullableHeap` which says that it is *safer*. I mimicked the other fields in `HTMLInputElement` so I initially wrote the wrapper as this. `MutNullableHeap` looks like a primitive interface provided by jsapi.